### DISCUSSION TITLE
Missing track: Expected (28802, 6, 2, 6) vs (28802, 6, 2, 5)

### DISCUSSION THREAD

**Discussion by lisadiez** [talmolab/sleap#1238] [Help!] [answered]
Hello!

We're using a bottom-up model to predict on videos containing 6 animals.

We have almost 1000 videos and are running into problems with analysis because some files contain 5 tracks only...

Do you have any insight as to why this happens? Even if an animal is not detected, shouldn't we always obtain 6 tracks (but one filled with NaNs?)

Thank you for your help, as always!

**Comment by roomrys** [comment, marked as answer]
Hi @lisadiez,

SLEAP constructs the occupancy matrix for the analysis file using all the tracks, but then removes all the empty tracks from the matrix to clean it up. 
https://github.com/talmolab/sleap/blob/81b43425e98ab43a155eb2f3a46910d51e73ca61/sleap/info/write_tracking_h5.py#L193-L214

This was probably a design under the assumptions that 
1. users may not remove empty tracks manually before exporting to the analysis file 
2. videos are self-contained trials (do not expect identities to last across videos)
 
A better design might be to remove all truly unused tracks before writing to an occupancy matrix and then create the occupancy matrix leaving tracks unusued in just that video untouched for easier identification of which track is which.

Would that work better in your case? I am assuming you want/need the tracks across videos?

Thanks,
Liezl

**Reply by lisadiez** [reply]
Yes, we need all tracks across videos!

I’m missing tracks in the prediction slp file as well, not only in the analysis file. Is that also intentional?

**Reply by roomrys** [reply]
The prediction slp file will only contain the tracks of those predictions (nothing more).
